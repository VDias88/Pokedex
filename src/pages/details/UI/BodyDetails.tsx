import React,{ useEffect } from 'react'
import { View,Text,FlatList,StyleSheet } from 'react-native'
import { usePkmStore } from '../../../mobx/pkmProvider'
import PkmStore from '../../../mobx/pkmStore';
import Abilities from '../UI/Abilities' 
import StatsList from './StatsList';
import MovesList from './MovesList';


type BodyDetailsProps={
    pkmName:string,
}

const BodyDetails:React.FC<BodyDetailsProps>=({pkmName})=>{
    const { pkmStore } = usePkmStore()
    useEffect(() => {
      pkmStore.updatePokemonDetail(pkmName)
    }, [])
    
    const setHeight = (height)=>{
        return `${height/10}m`
    }
    const setWeight = (weight)=>{
        return `${weight/10}kg`
    }
    const listAb=(abilities)=>{
        if(abilities === undefined){

        }else{
            const list=[]
            abilities.forEach(e=>{
                list.push(e.ability.name)
            })
            return list
        }  
    }
    return(
        <View testID="body-details" style={styles.bodyStyle}>
           <View testID="body-details-prop" style={styles.contHead}>
                <View >
                    <Text style={styles.headText} >{`${setHeight(pkmStore.pokemonDetail.height)}`}</Text>
                    <Text style={styles.headSubText}>Altura</Text>
                </View>
                <View>
                    <Text style={styles.headText} >{`${setWeight(pkmStore.pokemonDetail.weight)}`}</Text>
                    <Text style={styles.headSubText}>Peso</Text>
                </View>
           </View>
           <View>
               <Abilities abilities={listAb(pkmStore.pokemonDetail.abilities)}/>
           </View>
           <View>
                <StatsList  stats={pkmStore.pokemonDetail.stats} color={pkmStore.pokemonDetail?.types?.[0]?.type?.name}/>
           </View>
           <View>
                <MovesList />
           </View>
        </View>
    )
}
export default BodyDetails

const styles=StyleSheet.create({
    contBox:{
        fontSize:20
    },
    contHead:{
        flexDirection:'row',
        justifyContent:'space-evenly',
        paddingVertical:20
        
    },
    headText:{
        fontWeight:'bold',
        fontSize:20
    },
    headSubText:{
        alignSelf:'center'
    },
    bodyStyle:{
        height:'100%',
        
    }
})