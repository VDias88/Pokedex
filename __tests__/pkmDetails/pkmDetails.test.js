import React from 'react'
import { render } from '@testing-library/react-native'
import TopDetails from '../../src/pages/details/UI/TopDetails';
import BodyDetails from '../../src/pages/details/UI/BodyDetails';

let component = null

beforeAll(()=>{
    
   
})


describe('Pkm Details Tests',()=>{      
    it('Should render TopDetails correctly',()=>{//it.skip -- passsar teste
        const { queryByTestId }=render(<TopDetails />)
        const component=queryByTestId('top-details')
        expect(component).toBeTruthy()
    });

    it('should has a pokedex position',()=>{
        const { queryByTestId }=render(<TopDetails pokedexEntry={'25'}/>)
        const component=queryByTestId('pokedex-entry-number')
        expect(component.props.children).toBe('#025')
    });

    it('should render pokemo name',()=>{
        const pokemonName='pikachu'
        const { queryByTestId }=render(<TopDetails pokemonName={pokemonName}/>)
        const component=queryByTestId('pokemon-name')
        expect(component.props.children).toBe(pokemonName)
    });

    it('should has a background color prop',()=>{
        const pokemonTypeColor={
            main:'orange',
            accent:'#FFCA35'
        }

        const { queryByTestId }=render(<TopDetails pokemonTypeColor={pokemonTypeColor}/>)
        const component=queryByTestId('top-details')
        expect(component.props.style[1].backgroundColor).toBe(pokemonTypeColor?.main)
    });

    it('should render pokemo type badge',()=>{
        const pokemonTypes=['electric','normal']

        const { queryAllByTestId }=render(<TopDetails pokemonTypes={pokemonTypes}/>)
        const components=queryAllByTestId('pokemon-type')
        
        expect(components.length).toEqual(pokemonTypes.length)

        components.forEach((component,index)=>{
            expect(component.props.children).toBe(pokemonTypes[index])
        })
    });
    it('should render pokemo image',()=>{
        const url='pikachu'

        const { queryByTestId }=render(<TopDetails pkmImage={url}/>)
        const component=queryByTestId('pokemon-image')
        expect(component.props.source.uri).toBe(url)
    });
    it('should match snapshot <TopDetails />',()=>{
        const url='pikachu'
        const pokemonTypes=['electric','normal']
        const pokemonName='pikachu'
        const pokedexEntry=32
        const component=render(<TopDetails 
        pkmImage={url}
        pokemonName={pokemonName}
        pokedexEntry={pokedexEntry}
        pokemonTypes={pokemonTypes}
        />).toJSON()
        expect(component).toMatchSnapshot()
    });

    it('Shold render BodyDetails correctly',()=>{
        const { queryByTestId } = render (<BodyDetails /> )
        const component=queryByTestId('body-details')
        expect(component).toBeTruthy()
    });
    //renderizar prop
    it('Shold render pkmname prop',()=>{
        const pkmName="pikachu"
        const { queryByTestId } = render (<BodyDetails pkmName={pkmName}/> )
        const component=queryByTestId('body-details-prop')
        expect(component).toBeTruthy()
    })
})