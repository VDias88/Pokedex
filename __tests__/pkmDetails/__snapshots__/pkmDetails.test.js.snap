// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pkm Details Tests should match snapshot <TopDetails /> 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 20,
        "paddingBottom": 70,
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
  testID="top-details"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
        testID="pokedex-entry-number"
      >
        #032
      </Text>
      <Text
        style={
          Object {
            "fontSize": 32,
            "fontWeight": "bold",
          }
        }
        testID="pokemon-name"
      >
        pikachu
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.1)",
            "borderRadius": 20,
            "color": "white",
            "marginHorizontal": 2.5,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
        testID="pokemon-type"
      >
        electric
      </Text>
      <Text
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.1)",
            "borderRadius": 20,
            "color": "white",
            "marginHorizontal": 2.5,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
        testID="pokemon-type"
      >
        normal
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "uri": "pikachu",
      }
    }
    style={
      Object {
        "height": 120,
        "position": "absolute",
        "right": 0,
        "top": 30,
        "width": 120,
      }
    }
    testID="pokemon-image"
  />
</View>
`;
